Sources/TracingInstrumentation/Span.swift

```diff
@@ -121,34 +121,46 @@ public struct SpanAttributeKey<T>: Hashable, ExpressibleByStringLiteral {
 
 @dynamicMemberLookup
 public protocol SpanAttributeNamespace {
-    var attributes: SpanAttributes { get set }
+    /// Type that contains the nested attributes, e.g. HTTPAttributes which would contain `statusCode` and similar vars.
     associatedtype NestedAttributes: NestedSpanAttributesProtocol
+    
+    var attributes: SpanAttributes { get set }
 
-    subscript<T>(dynamicMember dynamicMember: KeyPath<SpanAttribute, SpanAttributeKey<T>>) -> SpanAttribute? { get set }
+    subscript<T>(dynamicMember dynamicMember: KeyPath<NestedAttributes, SpanAttributeKey<T>>) -> T? where T: SpanAttributeConvertible { get set }
 
     subscript<Namespace>(dynamicMember dynamicMember: KeyPath<SpanAttribute, Namespace>) -> Namespace
         where Namespace: SpanAttributeNamespace { get }
 }
 
 public protocol NestedSpanAttributesProtocol {
-    static var namespace: Self { get }
+    init()
+    static var __namespace: Self { get }
+}
+
+extension NestedSpanAttributesProtocol {
+    public static var __namespace: Self { .init() }
 }
```

Comment:
this simulates what we were doing before -- the namespace really was just a way to get "an instance" so we do this "manually" now; as structs are values this should remain cheap.